Update reverse-proxy.md (#5241)
nextcloud/all-in-one: reverse-proxy.md

# Reverse Proxy Documentation

> [!NOTE]  
> The maintainers of AIO noticed that this documentation could be improved to make it easier to follow. All contributions that improve this are very welcome!

> A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically a software service that can be thought of as the gateway of a web server. It enables computers on the internet to access a service or web-site in a [private subnet](https://en.wikipedia.org/wiki/Private_network) of that web server.

Reviewer: ```suggestion
A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is a software service that acts as a gateway between services and a client. It is commonly used to allow a client connected to the Internet to access a website located in the [private subnet](https://en.wikipedia.org/wiki/Private_network) of that web server.
```

**Please note:** Publishing the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get-a-valid-certificate-for-the-aio-interface).

> In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to <br>1. specify the port that AIO's integrated Apache container shall use <br>2. add a specific config to your web server or reverse proxy <br>3. modify the startup command a bit. <br>All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private network to receive unencrypted HTTP traffic from the reverse proxy. **Advice:** If you need HTTPS between Nextcloud and the reverse proxy because it is running on a different server in the same network, simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.

Reviewer: ```suggestion
In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to:
1. specify the port that AIO's integrated Apache container shall use
2. add a specific config to your web server or reverse proxy
3. modify the startup command a bit.
All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private network to receive unencrypted HTTP traffic from the reverse proxy.
> [!IMPORTANT] 
> If you need HTTPS between Nextcloud and the reverse proxy because it is running on a different server in the same network, simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.
```
> simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.

No, "simply" use port 443.... Nextcloud is exposed in HTTPS...

**Note:** The maintainers of AIO noticed that this documentation could be improved to make it easier to follow. All contributions that improve this are very welcome!

> A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically the entry point of a web server. It allows other computers to access the [private subnet](https://en.wikipedia.org/wiki/Private_network) of the web server over the Internet.

Author: sorry, my mistake, somehow my sentence was truncated:

"A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) is basically a software service that can be thought of as the gateway of a web server that enables computers on the internet to access a service or web-site in a [private subnet] of that web server.

> **Please note:** Deploying the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get-a-valid-certificate-for-the-aio-interface).

Reviewer: I think publishing is better than deploying because it is deploying the interface without publishing it.

Reviewer 2: ```suggestion
**Please note:** Publishing the AIO interface with a valid certificate to the public internet is **not** the goal of this documentation! If you need a valid certificate for the AIO interface, see [point 5](#5-optional-get-a-valid-certificate-for-the-aio-interface).
```

> In order to run Nextcloud behind a web server / reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to 1. specify the port that AIO's integrated Apache container shall use 2. add a specific config to your web server / reverse proxy 3. modify the startup command a bit. All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private server to receive unencrypted HTTP traffic from the reverse proxy. **Advice:** If you need HTTPS between Nextcloud and the reverse proxy because it is running on a different server in the same network, simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.

Reviewer 2: you could also use the <br> tag

> In order to run Nextcloud behind a web server / reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to 1. specify the port that AIO's integrated Apache container shall use 2. add a specific config to your web server / reverse proxy 3. modify the startup command a bit. All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private server to receive unencrypted HTTP traffic from the reverse proxy. **Advice:** If you need HTTPS between Nextcloud and the reverse proxy because it is running on a different server in the same network, simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.

Reviewer 2: ```suggestion
In order to run Nextcloud behind a web server or reverse proxy (like Apache, Nginx, Caddy, Cloudflare Tunnel and else), you need to <br>1. specify the port that AIO's integrated Apache container shall use <br>2. add a specific config to your web server or reverse proxy <br>3. modify the startup command a bit. <br>All examples below will use port `11000` as `APACHE_PORT`. This port will be exposed in the private network to receive unencrypted HTTP traffic from the reverse proxy. **Advice:** If you need HTTPS between Nextcloud and the reverse proxy because it is running on a different server in the same network, simply add another reverse proxy to the chain that runs on the same server like AIO and takes care of HTTPS proxying (most likely via self-signed certificates). Another option would be to create a VPN between the server that runs AIO and the server that runs the reverse proxy which takes care of encrypting the connection.
```

    reverse_proxy localhost:11000
}
```

> The Caddyfile is a text file called `Caddyfile` (no extension) which – if you should be running Caddy inside a container – should usually be created in the same location as your `docker-compose.yaml` file prior to starting the container.

Reviewer: Since compose files are used by [multiple backends](https://github.com/compose-spec/compose-spec#implementations), `compose.yaml` is more relevant. See:
https://github.com/compose-spec/compose-spec/blob/231b09c30d339e950c0da17fe5bdc793366b8fde/03-compose-file.md?plain=1#L11-L13

Reviewer 2: ```suggestion
The Caddyfile is a text file called `Caddyfile` (no extension) which – if you should be running Caddy inside a container – should usually be created in the same location as your `compose.yaml` file prior to starting the container.
```

1. Install the Cloudflare Tunnel on the same machine where AIO will be running on and point the Tunnel with the domain that you want to use for AIO to `http://localhost:11000`.<br>

> ⚠️ **Please note:** Look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.

Reviewer: ```suggestion
⚠️ **Please note:** look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.
```
consistent

```

> ⚠️ **Please note:** look in [here](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.

Reviewer: I find it weird, but I'm not a native English speaker so I'm not sure.

Reviewer 2: Also not sure but lets leave it like this for now

**Advice:** You may have a look at [this](https://github.com/nextcloud/all-in-one/discussions/588#discussioncomment-2811152) for a more complete example.

</details>

### Nginx-Proxy-Manager - NPM 🔀

<details>

<summary>click here to expand</summary>

> First, make sure the environmental variables `PUID` and `PGID` in the docker-compose.yaml file for NPM are either unset or set to `0`.

Reviewer: Same

Reviewer 2: ```suggestion
First, make sure the environmental variables `PUID` and `PGID` in the `compose.yaml` file for NPM are either unset or set to `0`.
```

});
```

> ⚠️ **Please note:** look into [this](#adapting-the-sample-web-server-configurations-below) to adapt the above example configuration.

Reviewer: Here it has not been changed?

</details>

## 2. Use this startup command 🚀

After adjusting your reverse proxy config, use the following command to start AIO:<br>

> (For a docker-compose.yml example, see the example further [below](#inspiration-for-a-docker-compose-file).)

Reviewer: ```suggestion
(For a `compose.yaml` example, see the example further [below](#inspiration-for-a-docker-compose-file).)
```
